### DISCUSSION TITLE
GIF materials not being shown correctly (custom module)

### DISCUSSION THREAD

**Discussion by ghost** [Earu/EasyChat#103] [General]
Here is my custom module: https://paste.bingner.com/paste/2o2x3

The module tries to parse a folder, then add any found .png or .gif files as emojis. It works perfectly for .png files, but the materials of .gif files fail to be shown in chat, instead showing a checkered purple and black icon meaning a missing texture.

What confuses me here is that the `gif_material` function returns the proper material, and using print shows that it should simply work.

Any idea what I'm doing wrong here?

**Comment by Earu** [comment]
Hi, if I understand correctly its not rendering .gifs ? I'm not sure this is even supported by gmod, could you try to use that same material and render it in a `HUDPaint` hook and see what happens?
